netty/src/main/java/io/grpc/netty/TlsOptions.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.netty;
+
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLEngine;
+
+/**
+ * TlsOptions contains different options users could choose. In a nutshell, it provides three main
+ * features users could customize:
+ * 1. choose different levels of peer verification by specifying |VerificationAuthType|
+ * 2. provide custom peer verification check by inheriting |verifyPeerCertificate|
+ * 3. change the trust CA certificate bundle by inheriting |getTrustedCerts|
+ */
+public abstract class TlsOptions {
```

Comment:
Sorry, I'd like to discuss a bit more about this here. Making it an interface seems a reasonable common practice, but the implication becomes slightly different. Before, I want users to set a `VerificationAuthType` at the construction time, and this can be expressed by making it a construction function of an abstract class; If using interface, users may need to read the comments to know they need to set `VerificationAuthType`. The method might be more like `setVerificationAuthType` instead of `getVerificationAuthType` if using interface. May I know your opinions?
Also, does `TlsConfig` sound like a better name for you?

Reply:
I was not suggesting to make it an interface; keep it as an abstract class. We use abstract classes so we can add methods to them later.

No, even if the user is implementing it, `getVerificationAuthType()` is a proper name for it. I don't think that would cause a problem for Java programmers, especially those that would be implementing this.

The other option is to guarantee that the verification type does not change over time and make `getVerificationAuthType()` final. The problem now is that it can be overridden but it is also required for the constructor.

Reply:
SG. Updated accordingly. 
Regarding the name of this class, would `TlsConfig` sound better?